trestle/core/markdown/markdown_processor.py

```diff
@@ -84,3 +85,25 @@ def fetch_value_from_header(self, md_path: pathlib.Path, key: str) -> Optional[s
             value = header[key]
 
         return value
+
+    def cull_headings(self, tree: MarkdownNode, cull_list: List[str], flexible: bool = True) -> None:
+        """Cull contents from the tree that match list of headings."""
+
+        def should_cull(header: str, clean_list: List[str], flexible: bool) -> bool:
+            """Determine if this header is in the cull list."""
+            clean_header = spaces_and_caps_to_lower_single_spaces(header) if flexible else header
+            for item in clean_list:
+                if item in clean_header:
+                    return True
+            return False
+
+        clean_list: List[str] = [
+            spaces_and_caps_to_lower_single_spaces(item) for item in cull_list
+        ] if flexible else cull_list
+        new_list: List[MarkdownNode] = []
+        for subnode in tree.subnodes:
+            # if this isn't culled check its subnodes and add to list of new nodes
+            if not should_cull(subnode.key, clean_list, flexible):
+                self.cull_headings(subnode, clean_list, flexible)
+                new_list.append(subnode)
+        tree.subnodes = new_list
```

Comment:
There is an issue with this function. First, I assume that you would like to delete a section from the markdown including subsections of that section. I.e.
given 
```
# Heading 1
## Subheading 1.1
### Subsubheading 1.1.1
# The other heading 2
```
> cull_headings("# Heading 1")

Will result in entire section being deleted:
```
# The other heading 2
```
If you would like to delete only a single section and NOT it is children then there are more problems...

Assuming above, the issue with current function is that it will not delete things from the parent nodes and content of the node.

The logic here should be to traverse tree and once the node with matching heading is found, this node contents should be deleted from all parent nodes and parent's content.

Reply:
reverted the change